Gems/Blast/Editor/Scripts/blast_asset_builder.py

```diff
@@ -0,0 +1,281 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+"""
+This a Python Asset Builder script examines each .blast file to see if an
+associated .fbx file needs to be processed by exporting all of its chunks
+into a scene manifest
+
+This is also a SceneAPI script that executes from a foo.fbx.assetinfo scene
+manifest that writes out asset chunk data for .blast files
+"""
+import os, traceback, binascii, sys, json, pathlib
+import azlmbr.math
+import azlmbr.asset
+import azlmbr.asset.entity
+import azlmbr.asset.builder
+import azlmbr.bus
+
+#
+# Python Asset Builder
+#
+busId = azlmbr.math.Uuid_CreateString('{D4FA20E3-8EF4-44A3-A045-AAE6C1CCAAAB}', 0)
+jobKeyName = 'Blast Chunk Assets'
+
+def log_exception_traceback():
+    exc_type, exc_value, exc_tb = sys.exc_info()
+    data = traceback.format_exception(exc_type, exc_value, exc_tb)
+    print(str(data))
+
+def raise_error(message):
+    print (f'ERROR - {message}');
+    raise RuntimeError(f'[ERROR]: {message}');
+
+# creates a single job to compile for each platform
+def get_source_fbx_filename(request):
+    fullPath = os.path.join(request.watchFolder, request.sourceFile)
+    basePath, filePart = os.path.split(fullPath)
+    filename = os.path.splitext(filePart)[0] + '.fbx'
+    filename = os.path.join(basePath, filename)
+    return filename
+
+def create_jobs(request):
+    fbxSidecarFilename = get_source_fbx_filename(request)
+    if (os.path.exists(fbxSidecarFilename) is False):
+        print('[WARN] Sidecar FBX file {} is missing for blast file {}'.format(fbxSidecarFilename, request.sourceFile))
+        return azlmbr.asset.builder.CreateJobsResponse()
+
+    # see if the FBX file already has a .assetinfo source asset, if so then do not create a job
+    establishedAssetInfo = f'{fbxSidecarFilename}.assetinfo';
+    if (os.path.exists(establishedAssetInfo)):
+        response = azlmbr.asset.builder.CreateJobsResponse()
+        response.result = azlmbr.asset.builder.CreateJobsResponse_ResultSuccess
+        return response
```

Comment:
Should it not at least establish a Source Dependency on that sidecar file so that if it gets deleted, or created, this job reruns?

Reply:
Actually this one is to explicit NOT have a source dependency with a foo.fbx file that already has an Editor authored foo.fbx.assetinfo file since the blast scene has already been set up.